boringssl-static/pom.xml

```diff
@@ -962,6 +974,248 @@
         </plugins>
       </build>
     </profile>
+    <!-- Profile to cross-compile for mac m1 -->
+    <profile>
+      <id>osx-m1-cross-compile</id>
+
+      <properties>
+        <linkStatic>true</linkStatic>
+        <osxCrossCompile>true</osxCrossCompile>
+        <macOsxDeploymentTarget/>
+        <cmakeOsxDeploymentTarget/>
+        <cflags>-target arm64-apple-macos11 -Werror -fno-omit-frame-pointer -fvisibility=hidden -Wunused -Wno-unused-value -O3</cflags>
```

Comment:
Not in this PR, but at some point I think it would be good to have `-Werror` be controlled by a property, so people can turn it off if they have a different compiler or compiler version.